src/en/hiperdex/src/eu/kanade/tachiyomi/extension/en/hiperdex/Hiperdex.kt

```diff
@@ -59,32 +55,24 @@ class Hiperdex :
         addRandomUAPreferenceToScreen(screen)
     }
 
-    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
-        return client.newCall(searchRequest(page, query, filters))
-            .asObservableSuccess().map { response ->
-                val document = response.asJsoup()
-                val anchor = document.getElementById("loop-content")
-                if (anchor == null) {
-                    MangasPage(emptyList(), false)
-                } else {
-                    val mangaList = mutableListOf<SManga>()
-                    val elementsList = anchor.select(".page-listing-item")
-                    for (element in elementsList) {
-                        mangaList.add(
-                            SManga.create().apply {
-                                val imageElement = element.selectFirst("img")!!
-                                val linkElement = element.selectFirst("a")!!
-
-                                setUrlWithoutDomain(linkElement.attr("href"))
-                                title = linkElement.attr("title")
-                                thumbnail_url = imageFromElement(imageElement)
-                            },
-                        )
-                    }
-
-                    MangasPage(mangaList, false)
-                }
+    override fun searchMangaSelector() = "#loop-content div.page-listing-item"
+
+    override val searchMangaUrlSelector = "div.item-thumb a"
+
+    override fun searchMangaFromElement(element: Element): SManga {
+        val manga = SManga.create()
+
+        with(element) {
+            selectFirst(searchMangaUrlSelector)!!.let {
+                manga.setUrlWithoutDomain(it.attr("abs:href"))
+                manga.title = it.attr("title")
             }
+            selectFirst("img")?.let {
+                manga.thumbnail_url = imageFromElement(it)
+            }
+        }
+
+        return manga
     }
```

Comment:
The original `searchMangaUrlSelector` already works. Keep the override of `searchMangaSelector()` and remove the rest.